spec/javascripts/modules/global-bar.spec.js

```diff
@@ -0,0 +1,157 @@
+describe('Global bar module', function () {
+  'use strict'
+
+  var globalBar
+  var element
+
+  beforeEach(function () {
+    globalBar = new GOVUK.Modules.GlobalBar()
+
+    document.cookie = 'global_bar_seen=; Path=/; Expires=Thu, 01 Jan 1970 00:00:01 GMT;';
+  })
+
+  describe('global banner default', function () {
+    beforeEach(function() {
+      element = $('\
+      <div id="global-bar" data-module="global-bar">\
+        <a href="/register-to-vote" class="govuk-link js-call-to-action">Register to Vote</a>\
+        <a href="#hide-message" class="govuk-link dismiss" role="button" aria-controls="global-bar">Hide message</a>\
+      </div>')
+
+      document.cookie = 'global_bar_seen=; Path=/; Expires=Thu, 01 Jan 1970 00:00:01 GMT;';
+    })
+
+    it('sets basic global_bar_seen cookie if not already set', function () {
+      expect(GOVUK.getCookie("global_bar_seen")).toBeNull()
+
+      globalBar.start(element)
+
+      expect(parseCookie(GOVUK.getCookie("global_bar_seen")).count).toBe(0)
```

Comment:
Shouldn't this be 1?

Reply:
It's unlikely this would happen as the initialise code should handle this case, but if the global-bar logic kicks in and the cookie is null or not what is expected, it sets a value of `{count: 0, version: 0}`

https://github.com/alphagov/static/pull/1965/files#diff-146dfe5256358c4209bb150ac180fabeR17